### ISSUE TITLE
TCH010 False positive when one argument is a TypeVar.

### ISSUE THREAD

**Issue by randolf-scholz** [astral-sh/ruff#11965]
[Rule TCH010](https://docs.astral.sh/ruff/rules/runtime-string-union/) should not apply when either argument is a TypeVar:

```python
from typing import TypeVar
X =TypeVar("X")
X | "test"  # no RuntimeError, because TypeVar implements __or__
```

This matters for instance for recursive type aliases:

```python
NestedMapping: TypeAlias = Mapping[K, V | "NestedMapping[K,V]"]
```

Here, writing `Mapping[K, "V | NestedMapping[K,V]"]` instead would cause errors at runtime, because the type-alias will only expect one argument:

```python
from typing import TypeVar, Mapping, TypeAlias
K = TypeVar("K")
V = TypeVar("V")
NestedMapping: TypeAlias = Mapping[K, V | "NestedMapping[K,V]"]
NestedMapping2: TypeAlias = Mapping[K, "V | NestedMapping2[K,V]"]
NestedMapping[str, float]  # Mapping[str, Union[float, ForwardRef('NestedMapping[K,V]')]]
NestedMapping2[str, float]  # TypeError: Too many arguments
```

**Comment by AlexWaygood** [comment]
I feel like the only way we could implement this is by special-casing `TypeVar` and certain other special types from the `typing` module. But your use case is fairly persuasive, so I guess that's what we'll have to do :/

**Comment by randolf-scholz** [comment]
This probably can be closed with `wont-fix`, as this resolves itself with PEP 695 style type aliases.
